torchio/data/images.py

```diff
@@ -41,7 +41,7 @@ def __init__(self, name: str, path: TypePath, type_: str, **kwargs):
         self.name = name
         self.path = self._parse_path(path)
         self.type = type_
-        self.kwargs = kwargs
```

Comment:
This attribute is referenced in `ImagesDataset` so I guess it should stay around.

Reply:
Also, if there `kwargs` has many items, users would need many lines of code to add the parameters to the sample. This way, it's just a single line.

Reply:
I missed the reference in ImageDataset, but what do you mean with that your need many likes of code to add parameters to a sample? Do you mean, like when you'd like a list of labels from an ImageDataset instance?

Reply:
And what if Subject inherits from 'dict' instead of 'list'? Then you could easily iterate though Subject.keys()

Reply:
I'm not sure that would be necessary. Why not just keeping `kwargs`?

Reply:
Hmm yea you are right it's not really necessary, maybe it's just a matter of taste then. Accessing Subject.kwargs['key'] feels a bit awkward as I would say the keyword arguments are attributes of the Subject. However, it works perfectly fine this way so it is a good sollution ;) I'll just work with it like this

Reply:
Maybe there's a misunderstanding. I propose keeping the kwargs attribute **and** doing the `Subject.__dict__.update(kwargs)`. So you can still do `subject.diagnose` or, if needed, `subject.kwargs['diagnose']`.

If `Subject` inherits from `dict`, I guess then there's no need for `Image` to have a `name`. Maybe that makes more sense. But what would be the arguments of `Subject`? A dictionary similar to the `sample_dict` would make sense, although it might be a bit more verbose than the current approach.

Reply:
Having both options of `subject.kwargs['diagnose']` and `subject.diagnose` would be a solution. It would also be really easy to add.

As I'm reading [online](https://stackoverflow.com/questions/2033150/subclassing-dict-should-dict-init-be-called), I am starting to doubt a little bit on whether it makes sense to inherrit from `dict`. Apparently if you'd do, then it would be good practice to make the `Subject.__init__` function similar to how `dict.__init__` works. I think that would mean that the current initialization should be altered such that `Subject.__init__` should accept `key:value` pairs instead of positional arguments. Which would mean that it break all code written with previous versions right?

Reply:
Yes. That's what I propose in my [comment below](https://github.com/fepegar/torchio/pull/115#issuecomment-609696750). Let's continue the discussion there, please.

Don't worry about breaking the code: "[Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable](https://semver.org/#spec-item-4)".

If it makes the library better, I think some disruption is acceptable in this experimental phase.